pkg/provider/provider.go

```diff
@@ -136,9 +183,21 @@ func ConfigureProvider(s *schema.ResourceData) (interface{}, error) {
 	oauthAccessToken := s.Get("oauth_access_token").(string)
 	region := s.Get("region").(string)
 	role := s.Get("role").(string)
+	oauthRefreshToken := s.Get("oauth_refresh_token").(string)
+	oauthClientID := s.Get("oauth_client_id").(string)
+	oauthClientSecret := s.Get("oauth_client_secret").(string)
+	oauthEndpoint := s.Get("oauth_endpoint").(string)
+	oauthRedirectURL := s.Get("oauth_redirect_url").(string)
 
-	dsn, err := DSN(account, user, password, browserAuth, privateKeyPath, oauthAccessToken, region, role)
+	if oauthRefreshToken != "" {
```

Comment:
```suggestion
	if oauthRefreshToken {
```

Reply:
I think we want to keep `if oauthRefreshToken != ""` since `if oauthRefreshToken` won't compile for strings.

Reply:
if i change to "if oauthRefreshToken {" i get this error non-boolean condition in if statementcompilerInvalidCond, keeping it "if oauthRefreshToken != """